.github/workflows/docker_release.yml

```diff
@@ -0,0 +1,74 @@
+name: Release Docker images
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - massi/docker
+    tags:
+      - v*
+
+env:
+  DOCKER_REPO_NAME: deepset/haystack
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USER }}
+          password: ${{ secrets.DOCKER_HUB_TOKEN }}
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: $DOCKER_REPO_NAME
+
+      - name: Build base images
+        uses: docker/bake-action@v2
+        env:
+          IMAGE_TAG_SUFFIX: ${{ steps.meta.outputs.version }}
+        with:
+          workdir: docker
+          targets: base
+          push: true
+
+      - name: Build Haystack images
+        uses: docker/bake-action@v2
+        env:
+          IMAGE_TAG_SUFFIX: ${{ steps.meta.outputs.version }}
+          BASE_IMAGE_TAG_SUFFIX: ${{ steps.meta.outputs.version }}
+        with:
+          workdir: docker
+          targets: api
+          push: true
+
+      - name: Release Haystack images
+        uses: docker/bake-action@v2
+        if: startsWith(github.ref, 'refs/tags/')
```

Comment:
Not sure about this line, what is it doing? Are we not building the `api-latest` in some cases?

Reply:
We only build `api-latest` during a proper release and not when merging in `main`. But now I see a potential problem with bugfix releases, let me think about this.

Reply:
Ok there was indeed a problem with the bugfix releases, now I compare the current version being built with the latest Haystack and only tag if the current one is greater